packages/react-table/src/components/TableComposable/examples/ComposableTableActions.tsx

```diff
@@ -0,0 +1,163 @@
+/* eslint-disable no-console */
+import React from 'react';
+import {
+  ButtonVariant,
+  DropdownToggle,
+  ToggleGroup,
+  ToggleGroupItem,
+  ToggleGroupItemProps
+} from '@patternfly/react-core';
+import {
+  TableComposable,
+  Thead,
+  Tr,
+  Th,
+  Tbody,
+  Td,
+  CustomActionsToggleProps,
+  ActionsColumn,
+  IAction
+} from '@patternfly/react-table';
+
+interface Repository {
+  name: string;
+  branches: string;
+  prs: string;
+  workspaces: string;
+  lastCommit: string;
+}
+
+type ExampleType = 'defaultToggle' | 'customToggle';
+
+export const ComposableTableActions: React.FunctionComponent = () => {
+  // In real usage, this data would come from some external source like an API via props.
+  const repositories: Repository[] = [
+    { name: 'one', branches: 'two', prs: 'a', workspaces: 'four', lastCommit: 'five' },
+    { name: 'a', branches: 'two', prs: 'k', workspaces: 'four', lastCommit: 'five' },
+    { name: 'p', branches: 'two', prs: 'b', workspaces: 'four', lastCommit: 'five' },
+    { name: '4', branches: '2', prs: 'b', workspaces: 'four', lastCommit: 'five' },
+    { name: '5', branches: '2', prs: 'b', workspaces: 'four', lastCommit: 'five' }
+  ];
+
+  const columnNames = {
+    name: 'Repositories',
+    branches: 'Branches',
+    prs: 'Pull requests',
+    workspaces: 'Workspaces',
+    lastCommit: 'Last commit'
+  };
+
+  // This state is just for the ToggleGroup in this example and isn't necessary for TableComposable usage.
+  const [exampleChoice, setExampleChoice] = React.useState<ExampleType>('defaultToggle');
+  const onExampleTypeChange: ToggleGroupItemProps['onChange'] = (_isSelected, event) => {
+    const id = event.currentTarget.id;
+    setExampleChoice(id as ExampleType);
+  };
+
+  const customActionsToggle = (props: CustomActionsToggleProps) => (
+    <DropdownToggle onToggle={props.onToggle} isDisabled={props.isDisabled}>
+      Actions
+    </DropdownToggle>
+  );
+
+  const defaultActions = (repo: Repository): IAction[] => [
```

Comment:
Should we make the return type `IAction[] | null`? Otherwise we can get a type error for the below assignment `rowActions = null`

Reply:
Well, the function never returns null, it's only the `rowActions` variable that could either contain actions or be null. I'm honestly surprised we aren't getting a type error when setting it to null, to be honest. I'll put an explicit annotation on the variable.